providers/zohocrm/read_test.go

```diff
@@ -0,0 +1,106 @@
+package zohocrm
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/amp-labs/connectors"
+	"github.com/amp-labs/connectors/common"
+	"github.com/amp-labs/connectors/test/utils/mockutils/mockserver"
+	"github.com/amp-labs/connectors/test/utils/testroutines"
+	"github.com/amp-labs/connectors/test/utils/testutils"
+)
+
+func TestRead(t *testing.T) { // nolint:funlen,gocognit,cyclop
+	t.Parallel()
+
+	contactsResponse := testutils.DataFromFile(t, "contacts.json")
+	callsResponse := testutils.DataFromFile(t, "calls.json")
+	unsupportedResponse := testutils.DataFromFile(t, "unsupportedread.json")
+
+	tests := []testroutines.Read{
+		{
+			Name:         "Object Name is required",
+			Server:       mockserver.Dummy(),
+			ExpectedErrs: []error{common.ErrMissingObjects},
+		},
+		{
+			Name:         "At least one field is required",
+			Input:        common.ReadParams{ObjectName: "deals"},
+			Server:       mockserver.Dummy(),
+			ExpectedErrs: []error{common.ErrMissingFields},
+		},
+		{
+			Name:  "Unsupported object",
+			Input: common.ReadParams{ObjectName: "arsenal", Fields: connectors.Fields("assistant")},
+			Server: mockserver.Fixed{
+				Setup:  mockserver.ContentJSON(),
+				Always: mockserver.ResponseString(http.StatusBadRequest, string(unsupportedResponse)),
```

Comment:
```suggestion
				Always: mockserver.Response(http.StatusBadRequest, unsupportedResponse),
```
There is an option for `[]byte`. Please, use the same for other occurences in this PR